ml-agents/mlagents/trainers/saver/saver.py

```diff
@@ -0,0 +1,28 @@
+# # Unity ML-Agents Toolkit
+import abc
+
+
+class BaseSaver(abc.ABC):
+    """This class is the base class for the Saver"""
+
+    def __init__(self):
+        """
+        TBA
+        """
+        pass
+
+    @abc.abstractmethod
+    def register(self, module):
```

Comment:
We could make the  type for register `Union[Policy, Optimizer]`

Reply:
My idea was to keep it flexible so that if one day we have some module other than Policy/Optimizer or maybe we only want to save a policy without optimizer (not sure what the case would be though) we can handle it easily.
If those are very unlikely to happen I can change it.